### PR TITLE
Bump rancher webhook to v0.3.5-rc3

### PR DESCRIPTION
**Pull request by KevinJoiner**

### CODE REVIEW

**Review by MbolotSuse — APPROVED** [review]

**Review by kinarashah — APPROVED** [review]

**Comment by kinarashah** [comment]
Merging this to unblock agent customization feature testing, and it has 2 approvals.

**Pull request merged**
